pythonmaths/arithmetic.py

```diff
@@ -52,7 +52,10 @@ def divide(x: int | float, y: int | float) -> float:
     >>> arithmetic.divide(5, 2)
         2.5
     """
-    return x / y
+    try:
+        return x / y
+    except ZeroDivisionError as e:
```

Comment:
except ZeroDivisionError as e:
        raise ZeroDivisionError("Cannot divide by zero, why would you even try that?") from e